scripts/do-release.sh

```diff
@@ -21,13 +21,20 @@ function create_release() {
     local project="$1"
     local target="$2"
     local files=( "${@:3}" )
+    local latest=false
     local prerelease=false
 
     [[ -z "${semver[pre]}" ]] || prerelease=true
 
+    # Mark as latest release only when we're NOT pre releasing, and we're on the same major.minor as the latest released tag
+    [[ -n "${semver[pre]}" ]] || \
+        [[ $(git tag -l --sort=-v:refname "v*" | head -1 | cut -f1-2 -d.) != "v${semver['major']}.${semver['minor']}" ]] || \
+        latest=true
```

Comment:
It’s not obvious that this works for new releases (but it does, thanks to the pre-releases). However it means that after the first pre-release in a new branch, releases on the previous branch won’t be identified as the latest, even if they come before the first actual release on the new branch. We want to check whether the new version would be the newest version, ignoring pre-releases; something like

    newver="${semver['major']}.${semver['minor']}.${semver['patch']}"
    if [ "$((git tag -l --sort=-v:refname "v*" | grep -v -- -; echo "$newver") | sort -rV | head -n 1)" = "$newver" ]; then
        latest=true
    fi

Reply:
You're right, good catch :grinning: 

I updated and added tests to capture this case, and I used a similar approach to what you proposed (as your proposal would also mark pre-releases as latest).